src/execution/compiler/operator/seq_scan_translator.cpp

```diff
@@ -0,0 +1,219 @@
+  // Generate tuple at a time scan condition
+  auto predicate = op_->GetScanPredicate();
+  auto cond_translator = TranslatorFactory::CreateExpressionTranslator(predicate.Get(), codegen_);
+  ast::Expr *cond = cond_translator->DeriveExpr(this);
+  builder->StartIfStmt(cond);
+}
+
+void SeqScanTranslator::GenTVIClose(execution::compiler::FunctionBuilder *builder) {
+  // Close iterator
+  ast::Expr *close_call = codegen_->OneArgCall(ast::Builtin::TableIterClose, tvi_, true);
+  builder->Append(codegen_->MakeStmt(close_call));
+}
+
+void SeqScanTranslator::GenTVIReset(execution::compiler::FunctionBuilder *builder) {
+  // Reset iterator
+  ast::Expr *reset_call = codegen_->OneArgCall(ast::Builtin::TableIterReset, tvi_, true);
+  builder->Append(codegen_->MakeStmt(reset_call));
+}
+
+bool SeqScanTranslator::IsVectorizable(const terrier::parser::AbstractExpression *predicate) {
+  // TODO(Amadou): Does not currently work with negative numbers.
+  // Implement once that's fixed.
+  return false;
+}
+
+void SeqScanTranslator::GenVectorizedPredicate(FunctionBuilder *builder,
+                                               const terrier::parser::AbstractExpression *predicate) {
+  if (predicate->GetExpressionType() == terrier::parser::ExpressionType::CONJUNCTION_AND) {
+    GenVectorizedPredicate(builder, predicate->GetChild(0).Get());
+    GenVectorizedPredicate(builder, predicate->GetChild(1).Get());
+  } else if (TranslatorFactory::IsComparisonOp(predicate->GetExpressionType())) {
+    auto left_cve = dynamic_cast<const terrier::parser::ColumnValueExpression *>(predicate->GetChild(0).Get());
+    auto col_idx = pm_[left_cve->GetColumnOid()];
+    auto col_type = schema_.GetColumn(left_cve->GetColumnOid()).Type();
+    auto const_val = dynamic_cast<const terrier::parser::ConstantValueExpression *>(predicate->GetChild(1).Get());
+    auto trans_val = const_val->GetValue();
+    auto type = trans_val.Type();
+    ast::Expr *filter_val;
+    switch (type) {
+      case terrier::type::TypeId::TINYINT:
+        filter_val = codegen_->IntLiteral(terrier::type::TransientValuePeeker::PeekTinyInt(trans_val));
+        break;
+      case terrier::type::TypeId::SMALLINT:
+        filter_val = codegen_->IntLiteral(terrier::type::TransientValuePeeker::PeekSmallInt(trans_val));
+        break;
+      case terrier::type::TypeId::INTEGER:
+        filter_val = codegen_->IntLiteral(terrier::type::TransientValuePeeker::PeekInteger(trans_val));
+        break;
+      case terrier::type::TypeId::BIGINT:
+        filter_val =
+            codegen_->IntLiteral(static_cast<int32_t>(terrier::type::TransientValuePeeker::PeekBigInt(trans_val)));
+        break;
+      default:
+        UNREACHABLE("Impossible vectorized predicate!");
+    }
+    ast::Expr *filter_call = codegen_->PCIFilter(pci_, predicate->GetExpressionType(), col_idx, col_type, filter_val);
+    builder->Append(codegen_->MakeStmt(filter_call));
+  }
```

Comment:
What's supposed to happen if neither of the if statements match?  Right now we implicitly return, but that seems like a mistake rather than design.